lib/utils/manager/download_manager.dart

```diff
@@ -0,0 +1,226 @@
+          .close();
+      eventIdMapDownloadFileInfo.remove(eventId);
+    }
+  }
+
+  void initDownloadFileInfo(
+    Event event,
+    void Function(Either<Failure, Success>) onEvent,
+  ) {
+    final streamController = StreamController<Either<Failure, Success>>();
+    eventIdMapDownloadFileInfo[event.eventId] = DownloadFileInfo(
+      eventId: event.eventId,
+      cancelToken: CancelToken(),
+      downloadStateStreamController: streamController,
+      streamSubscription: streamController.stream.listen((data) {
+        onEvent(data);
+        data.fold(
+          (left) {
+            Logs().e(
+              'DownloadManager::initDownloadFileInfo(): $left',
+            );
+            _clear(event.eventId);
+          },
+          (right) {
+            if (right is DownloadedFileState) {
+              _clear(event.eventId);
+            }
+          },
+        );
+      }),
+    );
+  }
+
+  Future<void> _clear(String eventId) async {
+    try {
+      await eventIdMapDownloadFileInfo[eventId]
+          ?.downloadStateStreamController
+          .close();
+      await eventIdMapDownloadFileInfo[eventId]?.streamSubscription?.cancel();
+    } catch (e) {
+      Logs().e(
+        'DownloadManager::_clear(): $e',
+      );
+      rethrow;
+    } finally {
+      eventIdMapDownloadFileInfo.remove(eventId);
+    }
+  }
+
+  Future<void> download(
+    Uri uriPath, {
+    required String savePath,
+    required Event event,
+    onError,
+  }) async {
+    final streamController = eventIdMapDownloadFileInfo[event.eventId]
+        ?.downloadStateStreamController;
+    final cancelToken = eventIdMapDownloadFileInfo[event.eventId]?.cancelToken;
+    if (streamController == null || cancelToken == null) {
+      Logs().e(
+        'DownloadManager::download(): streamController or cancelToken is null',
+      );
+      return;
+    }
+    streamController.add(
+      const Right(
+        DownloadFileInitial(),
+      ),
+    );
+    if (!PlatformInfos.isWeb) {
+      workingQueue.addTask(
+        Task(
+          id: event.eventId,
+          runnable: () async {
+            try {
+              await mediaAPI.downloadFileInfo(
+                uriPath: uriPath,
+                savePath: savePath,
+                onReceiveProgress: (receive, total) async {
+                  if (total != 0 && receive == total) {
+                    await _handleDownloadFileDone(
+                      event,
+                      savePath,
+                      streamController,
+                    );
+                  } else {
+                    streamController.add(
+                      Right(
+                        DownloadingFileState(
+                          receive: receive,
+                          total: total,
+                        ),
+                      ),
+                    );
+                  }
+                },
+                cancelToken: cancelToken,
+              );
```

Comment:
separate in another method (same idea with `DownloadFileExtension::getFileInfo`)